### PR TITLE
Change target framework to netstandard2.1 to comply with EFCore repo target

### PR DESCRIPTION
**Pull request by curtisy1**
When using EntityFramework Core 5.x together with a .NET Core 3.x app this package cannot be installed because it targets net5.0 for EFCore 5 dependencies. The official EF Core repo [targets netstandard2.1](https://github.com/dotnet/efcore/blob/v5.0.3/src/EFCore.SqlServer/EFCore.SqlServer.csproj#L5) so I believe EntityFrameworkCore.Testing should too.

If this is intended, as an alternative, I think dual targeting should be possible but when I tried nuget couldn't restore any packages so it would probably require some more work

### CODE REVIEW

File: src/EntityFrameworkCore.Testing.Common/AsyncEnumerable.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@ public AsyncEnumerable(IEnumerable<T> enumerable)
             }
         }
 
-        public IAsyncEnumerator<T> GetAsyncEnumerator(CancellationToken cancellationToken = new())
+        public IAsyncEnumerator<T> GetAsyncEnumerator(CancellationToken cancellationToken = new CancellationToken())
```
Comment (rgvlee) [inline src/EntityFrameworkCore.Testing.Common/AsyncEnumerable.cs]: Please revert this change

File: src/EntityFrameworkCore.Testing.Common/AsyncEnumerator.cs [hunk 2]
```diff
@@ -14,12 +14,12 @@ public AsyncEnumerator(IEnumerable<T> enumerable)
 
         public ValueTask DisposeAsync()
         {
-            return new();
+            return new ValueTask();
```
Comment (rgvlee) [inline src/EntityFrameworkCore.Testing.Common/AsyncEnumerator.cs]: Please revert this change

File: src/EntityFrameworkCore.Testing.Common/AsyncEnumerator.cs [hunk 3]
```diff
@@ -14,12 +14,12 @@ public AsyncEnumerator(IEnumerable<T> enumerable)
...
         }
 
         public ValueTask<bool> MoveNextAsync()
         {
-            return new(_enumerator.MoveNext());
+            return new ValueTask<bool>(_enumerator.MoveNext());
```
Comment (rgvlee) [inline src/EntityFrameworkCore.Testing.Common/AsyncEnumerator.cs]: Please revert this change

File: src/EntityFrameworkCore.Testing.Common/Helpers/BaseMockedDbContextBuilder.cs [hunk 4]
```diff
@@ -12,7 +12,7 @@ public abstract class BaseMockedDbContextBuilder<TDbContext> : IMockedDbContextB
         /// <summary>
         ///     The create factory options.
         /// </summary>
-        protected readonly MockedDbContextFactoryOptions<TDbContext> Options = new();
+        protected readonly MockedDbContextFactoryOptions<TDbContext> Options = new MockedDbContextFactoryOptions<TDbContext>();
```
Comment (rgvlee) [inline src/EntityFrameworkCore.Testing.Common/Helpers/BaseMockedDbContextBuilder.cs]: Please revert this change

**Review by rgvlee — CHANGES_REQUESTED** [review]
Thanks for the PR. I'm happy to merge this contribution once the following has been addressed:

- Please revert the language changes and use C# 9 for this branch
- The 2 nuspec files should also be updated to target the correct folders

Thanks Lee

File: src/EntityFrameworkCore.Testing.Common/AsyncEnumerable.cs (hunk 1 above)
Comment (curtisy1) [reply src/EntityFrameworkCore.Testing.Common/AsyncEnumerable.cs]: Unfortunately, [this is a C# 9 feature](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-9#fit-and-finish-features) and is not supported when targeting .NET Standard.

The only option would be to multi-target net5.0 and netstandard2.1 since they [seem to be interoperable](https://docs.microsoft.com/en-us/dotnet/standard/net-standard#when-to-target-net50-vs-netstandard) and one could use the net5.0 target only for the language features.

If you're fine with multi-targeting, I'll change the TargetFramework accordingly. Not sure how nice it plays with the nuspec files but it should just be a matter of specifying lib/netstandard2.1, right?

Comment (rgvlee) [reply src/EntityFrameworkCore.Testing.Common/AsyncEnumerable.cs]: My understanding was that language support was a compiler version rather than the runtime version concern. A quick check shows that I can build and package branch 3.x for efcore5 targeting .netstandard2.1, and have that working as expected in a .netcore3.1 app. But as I haven't really done much with .NET 5.0 yet I'm not 100% on the nuances.

RE nuspec, that's pretty much it. There is a nuspec for both the moq and nsub packages.

image

Comment (curtisy1) [reply src/EntityFrameworkCore.Testing.Common/AsyncEnumerable.cs]: You're right, it is a compiler version concern. The default language version for netstandard2.1 is C# 8 however, so it might need to be set to 9 explicitly (or my setup is just weird and one of the many sdks I have installed makes this a false positive error).

I've reverted the changes now and changed the nuspec target, so it should be good to merge now

Comment (rgvlee) [reply src/EntityFrameworkCore.Testing.Common/AsyncEnumerable.cs]: Language can be set in the project file.

image

Rider allows this to be set via project properties, not sure about other IDEs

image

Comment (curtisy1) [reply src/EntityFrameworkCore.Testing.Common/AsyncEnumerable.cs]: Fixed and set to language version 9 explicitly

**Review by rgvlee — APPROVED** [review]

**Comment by rgvlee** [comment]
Thanks Alex. I'll push a new release this week.
